lib/dicon/executor.ex

```diff
@@ -12,18 +12,20 @@ defmodule Dicon.Executor do
   Connects to the given authority, returning a term that identifies the
   connection.
   """
-  @callback connect(authority :: binary) :: {:ok, identifier} | {:error, term}
+  @callback connect(authority :: binary) :: {:ok, identifier} | {:error, binary}
 
   @doc """
   Executes the given `command` on the given connection.
   """
-  @callback exec(identifier, command :: char_list) :: :ok
+  @callback exec(identifier, command :: char_list) :: :ok | {:error, binary}
 
   @doc """
   Copies the local file `source` over to the destination `target` on the given
   connection.
   """
-  @callback copy(identifier, source :: char_list, target :: char_list) :: :ok
+  @callback copy(identifier, source :: char_list, target :: char_list) :: :ok | {:error, binary}
+
+  @type t :: %__MODULE__{module: module, ref: identifier}
```

Comment:
I like the idea of renaming `:ref`, and `:id` may be a good solution (even if it may not be super-clear). I'm thinking of `:instance` as well, but it sounds too object-oriented. If we can't come up with anything better than `:id`, then `:id` it is. :)

(btw, the problem is not `:id` per-se, but the `identifier` type which has an awful awful name)